pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -448,21 +445,41 @@ public void disableBroker() throws PulsarServerException {
      */
     @Override
     public synchronized void doLoadShedding() {
-        for (LoadSheddingStrategy strategy : loadSheddingPipeline) {
-            final Map<String, String> bundlesToUnload = strategy.findBundlesForUnloading(loadData, conf);
-            if (bundlesToUnload != null && !bundlesToUnload.isEmpty()) {
-                try {
-                    for (Map.Entry<String, String> entry : bundlesToUnload.entrySet()) {
-                        final String bundle = entry.getKey();
-                        final String broker = entry.getValue();
-                        adminCache.get(broker).namespaces().unloadNamespaceBundle(
-                                LoadManagerShared.getNamespaceNameFromBundleName(bundle),
-                                LoadManagerShared.getBundleRangeFromBundleName(bundle));
+        if (!LoadManagerShared.isUnloadDisabledInLoadShedding(pulsar)) {
+            if (getAvailableBrokers().size() > 1) {
+                // Remove bundles who have been unloaded for longer than the grace period from the recently unloaded
+                // map.
+                final Iterator<Map.Entry<String, Long>> recentlyUnloadedIterator = loadData.getRecentlyUnloadedBundles()
+                        .entrySet().iterator();
+                while (recentlyUnloadedIterator.hasNext()) {
```

Comment:
This part could be simplified by using a filter like: 

```java
loadData.getRecentlyUnloadedBundles().removeIf( e -> e.getValue() < TIMEOUT );
```

( don't look at the actual check above..)